File: ExampleApp/app/src/main/java/com/motiv/example/ViewPagerFragmentsAdapter.java
package com.motiv.example;

import androidx.fragment.app.*;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

public class ViewPagerFragmentsAdapter extends FragmentStatePagerAdapter {

    public ViewPagerFragmentsAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() {

        return 3;
    }

    @Override
    public Fragment getItem(int position) {

        switch (position) {
            case 0:
                return new UsersFragment();
            case 1:
                return new PostsFragment();
            case 2:
                return new PhotosFragment();
        }
        throw new RuntimeException();
    }

    @Override
    public java.lang.String getPageTitle(int position) {

        switch (position) {
            case 0:
                return "UsersFragment";
            case 1:
                return "PostsFragment";
            case 2:
                return "PhotosFragment";
        }
        return "";
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/ViewPagerFragmentsAdapter.kt
 
package com.motiv.example
import androidx.fragment.app.*

public class ViewPagerFragmentsAdapter(val fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    override fun getCount(): Int {
        return 3
    } override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return UsersFragment()
            1 -> return PostsFragment()
            2 -> return PhotosFragment()
        }
        throw RuntimeException()
    } override fun getPageTitle(position: Int): String {
        when (position) {
            0 -> return "UsersFragment"
            1 -> return "PostsFragment"
            2 -> return "PhotosFragment"
        }
        return ""
    }
}
